app/controllers/users_controller.rb

```diff
@@ -19,4 +27,19 @@ def change_password
       render json: { errors: { old_password: ["is incorrect"] } }, status: 400
     end
   end
+
+  private
+    def user
+      @user ||= User.find(params[:id])
+    end
+
+    def initialize_user_update_params
+      if @user_update_params
```

Comment:
i had to do some processing on the form params that come in to see whether or not to include the password in the params. but i only want to do this once (since the params don't change per request), so I cache the results. Make sense?